### ISSUE TITLE
[Managed Disks] diskAccessID casing issue

### ISSUE THREAD

**Issue by cmendible** [Azure/azure-rest-api-specs#14192]
Hi team,

I'm working to add support for Network Access Policy to Managed Disks via Terraform , and working on adding the Disk Access ID when the Network Policy is set to AllowPrivate.

The issue is that the Disk Access ID returned by the Managed Disk API has an incorrect casing of the resource group segment. For instance returning:

`/subscriptions/*******/resourceGroups/ACCTESTRG/providers/Microsoft.Compute/diskAccesses/accda`

instead of 

`/subscriptions/*******/resourceGroups/acctestRG/providers/Microsoft.Compute/diskAccesses/accda`

In other words the casing of the resourceGroups segment of the Resource ID, found in the diskAccessID field is returned all uppercase instead of the actual casing the RG was created with.

**Comment by jaylabell** [comment]
Hi Vaibhav, 

This has to GitHub issue deals with disk access, do you mind taking a look. 
Thanks.

**Comment by Vaibhav-Agar** [comment]
Hi @cmendible,
This is by design in the platform. As RGs is not case sensitive, hopefully, this is not causing an issue. Resolving.

**Comment by cmendible** [comment]
Sorry @Vaibhav-Agar can you take a closer look to this? The experience is not consistent and it  causes issue when comparing the DiskAccessID strings during unit tests. 

if you run: `az group create -l westeurope -n MyDisks` the result is:

``` json
{
  "id": "/subscriptions/*******/resourceGroups/MyDisks",
  "location": "westeurope",
  "managedBy": null,
  "name": "MyDisks",
  "properties": {
    "provisioningState": "Succeeded"
  },
  "tags": null,
  "type": "Microsoft.Resources/resourceGroups"
}
```

Then: `az disk-access create -g MyDisks -l westeurope -n MyDiskAccess` returns

``` json
{
  "id": "/subscriptions/d*******/resourceGroups/MyDisks/providers/Microsoft.Compute/diskAccesses/MyDiskAccess",
  "location": "westeurope",
  "name": "MyDiskAccess",
  "privateEndpointConnections": null,
  "provisioningState": "Succeeded",
  "resourceGroup": "MyDisks",
  "tags": null,
  "timeCreated": "2021-05-07T08:28:41.103874+00:00",
  "type": "Microsoft.Compute/diskAccesses"
}
```

and then: `az disk create -g MyDisks -n MyDisk --size-gb 10 --network-access-policy AllowPrivate --disk-access /subscriptions/*******/resourceGroups/MyDisks/providers/Microsoft.Compute/diskAccesses/MyDiskAccess` which results in:

``` json
{
  "burstingEnabled": null,
  "creationData": {
    "createOption": "Empty",
    "galleryImageReference": null,
    "imageReference": null,
    "logicalSectorSize": null,
    "sourceResourceId": null,
    "sourceUniqueId": null,
    "sourceUri": null,
    "storageAccountId": null,
    "uploadSizeBytes": null
  },
  "diskAccessId": "/subscriptions/*******/resourceGroups/MYDISKS/providers/Microsoft.Compute/diskAccesses/MyDiskAccess",
  "diskIopsReadOnly": null,
  "diskIopsReadWrite": 120,
  "diskMBpsReadOnly": null,
  "diskMBpsReadWrite": 25,
  "diskSizeBytes": 10737418240,
  "diskSizeGb": 10,
  "diskState": "Unattached",
  "encryption": {
    "diskEncryptionSetId": null,
    "type": "EncryptionAtRestWithPlatformKey"
  },
  "encryptionSettingsCollection": null,
  "extendedLocation": null,
  "hyperVGeneration": "V1",
  "id": "/subscriptions/*******/resourceGroups/MyDisks/providers/Microsoft.Compute/disks/MyDisk",
  "location": "westeurope",
  "managedBy": null,
  "managedByExtended": null,
  "maxShares": null,
  "name": "MyDisk",
  "networkAccessPolicy": "AllowPrivate",
  "osType": null,
  "propertyUpdatesInProgress": null,
  "provisioningState": "Succeeded",
  "purchasePlan": null,
  "resourceGroup": "MyDisks",
  "securityProfile": null,
  "shareInfo": null,
  "sku": {
    "name": "Premium_LRS",
    "tier": "Premium"
  },
  "supportsHibernation": null,
  "tags": {},
  "tier": "P3",
  "timeCreated": "2021-05-07T08:31:00.838429+00:00",
  "type": "Microsoft.Compute/disks",
  "uniqueId": "cf7f29bf-9bdd-4a55-bb44-17907647b253",
  "zones": null
}
```

As you can see `az disk-access create` returns the correct casing for the resource group while `az disk create` returns incorrect casing (all uppercase)

**Comment by vaibhavggarg** [comment]
I think you might have the wrong email address here.

Regards,
Vaibhav

**Comment by vaibhavggarg** [comment]
Hi,

I think you might have the wrong email address here.

Regards,
Vaibhav

**Comment by cmendible** [comment]
@Vaibhav-Agar can hyou echl this comment please: https://github.com/Azure/azure-rest-api-specs/issues/14192#issuecomment-834181859
